tests/run-pass/concurrency/thread_locals.rs

```diff
@@ -0,0 +1,54 @@
+// ignore-windows
+
+#![feature(thread_local)]
+
+use std::thread;
+
+#[thread_local]
+static mut A: u8 = 0;
+#[thread_local]
+static mut B: u8 = 0;
+static mut C: u8 = 0;
+
+unsafe fn get_a_ref() -> *mut u8 {
+    &mut A
+}
+
+struct Sender(*mut u8);
+
+unsafe impl Send for Sender {}
+
+fn main() {
+    let ptr = unsafe {
+        let x = get_a_ref();
+        *x = 5;
+        assert_eq!(A, 5);
+        B = 15;
+        C = 25;
+        Sender(&mut A)
+    };
```

Comment:
Please add a comment explaining that we take a ptr to a thread-local in one thread and use it in anotehr thread. Generally tests should have comments explaining their intent when it is not otherwise obvious (through function names or the file name of the test). It can be quite hard otherwise to figure out why the test was written the way it was.